heron/stmgr/src/cpp/manager/instance-server.cpp

```diff
@@ -413,15 +423,16 @@ void InstanceServer::DrainTupleStream(proto::stmgr::TupleStreamMessage* _message
 
 void InstanceServer::SendToInstance2(sp_int32 _task_id,
                                   proto::system::HeronTupleSet2* _message) {
-  sp_string instance_id = task_id_to_name[_task_id];
-
-  ConnectionBufferLengthMetricMap::const_iterator it =
-    connection_buffer_length_metric_map_.find(instance_id);
+  if (_message->has_data()) {
+    sp_string instance_id = task_id_to_name[_task_id];
 
-  if ( it != connection_buffer_length_metric_map_.end() )
-    connection_buffer_length_metric_map_
-      [instance_id]->scope("packets")->incr();
+    ConnectionBufferLengthMetricMap::const_iterator it =
+      connection_buffer_length_metric_map_.find(instance_id);
 
+    if ( it != connection_buffer_length_metric_map_.end() )
+      connection_buffer_length_metric_map_
+        [instance_id]->scope("packets")->incr_by(_message->data().tuples_size());
```

Comment:
do you use the same scope 'packets' but by two dimensions, one of num_tuples(), the other of tuples_size()? if i understand correctly, you are counting both count and size under one name?

Reply:
The tuples field `HeronTupleSet2` is actually a repeated field. So if I call the function `tuples_size()`, I am getting the size of the tuple array. On the other hand, the num_tuples field in `TupleStreamMessage` is just an integer that tells me how many tuples are in the serialized set field. So both functions just give me number of tuples and not size.